### PR TITLE
utils: `id_gen!` optimization

### PR DESCRIPTION
**Pull request by Paraworker**
The current id generation logic may return an ID that is already in the HashSet.

I've also simplified the usage of `id_gen!`.

### CODE REVIEW

File: src/utils/ids.rs [hunk 1]
```diff
@@ -1,36 +1,32 @@
...
-                            id += 1;
-                        }
-                        id += 1;
-                        Some(id)
-                    },
-                );
-                if let Ok(id) = new_id {
-                    break id;
+                while !id_set.insert(*counter) {
+                    *counter += 1;
```
Comment (Drakulix) [inline src/utils/ids.rs:18]: Looking at this again, this should probably be a wrapping_add, just like the next increment.

**Review by Drakulix — COMMENTED** [review]
Thanks! The module generation makes this indeed much cleaner to use.

File: src/utils/ids.rs (hunk 1 above)
Comment (Paraworker) [reply src/utils/ids.rs]: Oh thanks! Forgot the overflow panic in Rust.

Comment (Paraworker) [reply src/utils/ids.rs]: Done.

**Review by Drakulix — APPROVED** [review]
LGTM now. Thanks!

**Pull request merged**